internal/pkg/manifest/workload.go

```diff
@@ -87,6 +87,7 @@ var (
 // WorkloadManifest represents a workload manifest.
 type WorkloadManifest interface {
 	ApplyEnv(envName string) (WorkloadManifest, error)
+	WindowsCompatibility() error
```

Comment:
The `WindowsCompatibility` method seems to be only used within the manifest pkg. 

I think it's safe for us to remove this from the interface right? 
```suggestion
```

and in the if-statements below we can do:
```go
		if err := m.windowsCompatibility(); err != nil {
			return nil, err
		}
```